lib/reline/ansi.rb

```diff
@@ -60,8 +61,36 @@ def self.set_default_key_bindings(config)
       config.add_default_key_binding_by_keymap(:emacs, key, func)
     end
   end
+  
+  @@input = STDIN
+  def self.input=(val)
+    @@input = val
+  end
+
+  @@output = STDOUT
+  def self.output=(val)
+    @@output = val
+  end
 
   def self.set_default_key_bindings_terminfo(config)
+    
+    # put terminal in tx mode so that
+    # keys match terminfo's bindings
+    begin
+      @@output.write Reline::Terminfo.tigetstr('smkx')
+    rescue Reline::Terminfo::TerminfoError
+      # capname is undefined
+    end
+    
+    # return terminal to rx mode
```

Comment:
This comment needs to be fixed too.  It's not about "transmit" vs "receive".  Leaving keypad transmit mode doesn't mean the terminal goes into "receive" mode.

I think either of these would work fine:
* "Leave keypad transmit mode"
* "Return terminal to normal mode".

For more context, consider xterm: with xterm `smkx` turns into `\E[?1h` and `\E=`, which sets the cursor keys and keypad to application mode, whereas `rmkx` turns into `\E[?1l` and `\E>`, which sets the cursor keys and keypad to normal mode.